src/Chapter1.hs

```diff
@@ -537,7 +542,11 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: (Ord a) => a -> a -> a -> a
+mid x y z
+  | z >= max x y    = max x y
+  | z >= min x y    = z
+  | otherwise       = min x y
```

Comment:
Nice solution!
You see that you use `min` and `max` of x and y several times. You can move it into the variables under the `where` clause and reuse in each guard 👍🏼